src/lib.rs

```diff
@@ -384,6 +384,14 @@ impl DeviceHandle {
 	}
 }
 
+fn to_cstring_safe(ptr: *const std::os::raw::c_char) -> CString {
+	if ptr == std::ptr::null() {
+		CString::new("").expect("CString::new failed")
```

Comment:
One more thing, just my code style preference. Please `return` here if the pointer is null and drop the `else` branch, so it's goes like

```
if not_good {
    return default_value;
}
happy_path_continued..
```